plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```
@@ -119,16 +121,55 @@ export class HogTransformerService {
+                        const globals = this.createInvocationGlobals(event)
+                        const filterGlobals = convertToHogFunctionFilterGlobal(globals)
+
+                        // Check if function has filters - if not, always apply
+                        if (hogFunction.filters?.bytecode) {
+                            try {
+                                const filterResults = checkHogFunctionFilters({
+                                    hogFunction,
+                                    filterGlobals,
+                                    eventUuid: globals.event?.uuid,
+                                })
+
+                                // If filter didn't pass and there was no error, skip this transformation
+                                if (!filterResults.match && !filterResults.error) {
+                                    transformationsSkipped.push(transformationIdentifier)
+                                    results.push({
+                                        invocation: createInvocation(
+                                            {
+                                                ...globals,
+                                                inputs: {}, // Not needed as this is only for a valid return type
+                                            },
+                                            hogFunction
+                                        ),
+                                        metrics: filterResults.metrics,
+                                        logs: filterResults.logs,
+                                        error: null,
+                                        finished: true,
+                                    })
+                                    continue
+                                }
+                            } catch (error) {
+                                // Already logged in the utility, continue with transformation for backward compatibility
```

No silent catches. This could catch anywhere and just silently fail which is super bad. If the internal function handles the try catch then we don't need one here. If this is just for testing, it should still have some logging or a captureException call otherwise we will never know...

---

u r right. i removed the try catch here as its already handled in the underlying function